Upgrading to scons 4.0.1 breaks gradle build due to "scons.bat" reference in build.gradle

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
3.2 branch (custom build as of 6 Aug 2020)


**OS/device including version:**
Windows 10


**Issue description:**
After upgrading to Python 3.8.5 and scons 4.0.1 to comply with latest Godot "Compiling for Windows" instructions, the subsequent build failed on the gradle (Android) part:

`
C:\godot\platform\android\java>gradlew build
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details
> Task :lib:compileGodotNativeLibsDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':lib:compileGodotNativeLibsDebug'.
**> A problem occurred starting process 'command 'scons.bat''**

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 42s
39 actionable tasks: 1 executed, 38 up-to-date
`

The problem is caused by this [godot issue](https://github.com/godotengine/godot/issues/32270#). That "fix" appends ".bat" to the "scons" command line on Windows. There is no scons.bat in the latest version. Only scons.exe. Removing this code from build.gradle fixes the issue for me (not appending filetype to "scons" command line).

--- a/platform/android/java/build.gradle
+++ b/platform/android/java/build.gradle
@@ -22,8 +22,6 @@ allprojects {
 }
 
 ext {
-    sconsExt = org.gradle.internal.os.OperatingSystem.current().isWindows() ? ".bat" : ""
-
     supportedAbis = ["armv7", "arm64v8", "x86", "x86_64"]
     supportedTargets = ["release", "debug"]
 

--- a/platform/android/java/lib/build.gradle
+++ b/platform/android/java/lib/build.gradle
@@ -64,10 +64,42 @@ android {
             throw new GradleException("Invalid default abi: " + defaultAbi)
         }
 
+        // Find scons' executable path
+        File sconsExecutableFile = null
+        def sconsName = "scons"
+        def sconsExts = (org.gradle.internal.os.OperatingSystem.current().isWindows()
+            ? [".bat", ".exe"]
+            : [""])
+        logger.lifecycle("Looking for $sconsName executable path")
+        for (ext in sconsExts) {
+            String sconsNameExt = sconsName + ext
+            logger.lifecycle("Checking $sconsNameExt")
+
+            sconsExecutableFile = org.gradle.internal.os.OperatingSystem.current().findInPath(sconsNameExt)
+            if (sconsExecutableFile != null) {
+                // We're done!
+                break
+            }
+
+            // Check all the options in path
+            List<File> allOptions = org.gradle.internal.os.OperatingSystem.current().findAllInPath(sconsNameExt)
+            if (!allOptions.isEmpty()) {
+                // Pick the first option and we're done!
+                sconsExecutableFile = allOptions.get(0)
+                break
+            }
+        }
+
+        if (sconsExecutableFile == null) {
+            throw new GradleException("Unable to find executable path for the '$sconsName' command.")
+        } else {
+            logger.lifecycle("Found executable path for $sconsName: ${sconsExecutableFile.absolutePath}")
+        }
+
         // Creating gradle task to generate the native libraries for the default abi.
         def taskName = getSconsTaskName(buildType)
         tasks.create(name: taskName, type: Exec) {
-            executable "scons" + sconsExt
+            executable sconsExecutableFile.absolutePath
             args "--directory=${pathToRootDir}", "platform=android", "target=${releaseTarget}", "android_arch=${defaultAbi}", "-j" + Runtime.runtime.availableProcessors()
         }
 

